### PR TITLE
Add ability to update domain names

### PR DESCRIPTION
**Pull request by alex-yau-ttd**
- Implement functionality, including calls to admin, inserting audit trails, and hooking up to the UI
- Accept some sonarlint suggestions, such as using `string` instead of `String` (`Wrapper objects should not be used for primitive types (typescript:S1533)`)

https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/01a67c88-3646-48bc-a68b-120df84fb113

Notes
- Styling improvements are done in https://github.com/IABTechLab/uid2-self-serve-portal/pull/370

### CODE REVIEW

File: src/web/services/keyPairService.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export async function GetKeyPairs(participantId?: number) {
       `/participants/${participantId ?? 'current'}/keyPairs`
     );
     if (result.status === 200) {
-      return (result.data as KeyPairDTO[]).map(mapKeyPairDTOToModel);
+      return result.data.map(mapKeyPairDTOToModel);
```
Comment (alex-yau-ttd) [inline src/web/services/keyPairService.ts:19]: sonarlint suggestion

File: src/web/screens/clientSideIntegration.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          href='https://unifiedid.com/docs/guides/publisher-client-side'
+          rel='noreferrer'
+        >
+          Client-Side Integration Guide
+        </a>
+        .
+      </p>
+      <div className='content-container'>
+        <KeyPairsTable keyPairs={keyPairData ?? []} onAddKeyPair={handleAddKeyPair} />
+        {domainNames && (
```
Comment (alex-yau-ttd) [inline src/web/screens/clientSideIntegration.tsx:77]: the `domainNames` conditional here prevents the table from briefly rendering with `No Domain Names` before rendering with the actual data.

File: src/api/routers/participants/participantsRouter.ts [hunk 3]
```diff
@@ -457,7 +458,7 @@ export function createParticipantsRouter() {
 
       const traceId = getTraceId(req);
       const auditTrail = await insertManageApiKeyAuditTrail(
-        participant!,
+        participant,
```
Comment (alex-yau-ttd) [inline src/api/routers/participants/participantsRouter.ts:461]: sonarlint suggestion

Comment (lionell-pack-ttd) [reply src/api/routers/participants/participantsRouter.ts:461]: Yep - we've confirmed in an earlier check that participant isn't null to the ! isn't needed.

File: src/web/screens/clientSideIntegration.tsx [hunk 4]
```diff
@@ -35,13 +46,38 @@ function ClientSideIntegration() {
     }
   };
 
+  const handleUpdateDomainNames = async (updatedDomainNames: string[]) => {
+    try {
+      const response = await SetDomainNames(updatedDomainNames);
```
Comment (lionell-pack-ttd) [inline src/web/screens/clientSideIntegration.tsx:51]: These two functions only differing by first letter case makes it easy to misunderstand what's going on here. Could you find a different name for one of them?

Comment (alex-yau-ttd) [reply src/web/screens/clientSideIntegration.tsx]: fixed!

**Review by lionell-pack-ttd — APPROVED** [review]

**Pull request merged**